ScreechBot/index.js

```diff
@@ -0,0 +1,179 @@
+      command.action(msg);
+      msg.delete();
+    }
+  });
+
+});
+
+//Help text generation. Appears in the following format:
+//
+// Looks like {@user} is an idiot and needs guidence on the commands:
+//
+// Command: {keyword}
+// {helptext}
+//
+// Next command etc...
+//
+//  - msg: the message object from discord https://discord.js.org/#/docs/main/stable/class/Message
+function help(msg){
+  var reply = `Looks like ${msg.member} is an idiot and needs guidence on the commands: \n\n`;
+  commands.forEach(command => {
+    reply += "Command: " + command.keyword + `\n`;
+    reply += command.helptext + `\n\n`
+  });
+
+  msg.channel.send(reply);
+}
+
+//The bot will join the voice channel that the command giver is currently on and then proceed to play an audio file or stream.
+// - audio: The file or stream to play - see https://discord.js.org/#/docs/main/stable/class/VoiceConnection?scrollTo=play
+// - leave: boolean for if the bot should leave the voice channel after playing audio.
+function playAudio(audio, leave){
+  return function(msg){
+    if (msg.member.voice.channel) {
+      msg.member.voice.channel.join().then(connection =>{
+        const dispatcher = connection.play(audio);
+        if(leave){
+          dispatcher.on("finish", end => {msg.member.voice.channel.leave();});
+        }
+        dispatcher.on("error", console.error);
+      }).catch(err => console.log(err));
+    } else {
+      msg.reply('You need to join a voice channel first!');
+    }
+  }
+}
+
+// Connects the bot to the voice channel that the command giver is on
+function join(msg){
+  // Only try to join the sender's voice channel if they are in one themselves
+  if (msg.member.voice.channel) {
+    msg.member.voice.channel.join();
+  } else {
+    msg.reply('You need to join a voice channel for me to follow');
+  }
+}
+
+
+// Disconnects the bot from the voice channel that the command giver is on
+function leave(msg){
+  // Only try to leave the sender's voice channel if they are in one themselves
+  if (msg.member.voice.channel) {
+    msg.member.voice.channel.leave();
+    msg.reply('Fine I will get off!');
+  } else {
+    msg.reply('You need to join a voice channel I am in to tell me to leave!');
+  }
+}
+
+// Sets the bot into a watchloop for the current voice channel that the command giver is on
+function watch(msg){
+  if (msg.member.voice.channel) {
+    channelWatchLoop(msg);
+  } else {
+    msg.reply('You need to join a voice channel first!');
+  }
+}
+
+
+// Loops over watching the voice channel of the command giver. emits voiceChat events when voice channel state changes.
+function channelWatchLoop(msg){
+  var voiceChannel = msg.member.voice.channel;
+  voiceChannel.join()
+    .then(connection => { // Connection is an instance of VoiceConnection
+      msg.reply('I have successfully connected to the channel and am now stalking you!');
+      var old_members = voiceChannel.members.array().length;
+      var continueLoop = true;
```

Comment:
I still think it's not very descriptive. Looks like the condition is set to false when it's the only one in the chat. So something like chatHasMultipleMembers or chatSessionHasNotEnded would be nice. However I don't think this is enough of a reason to block this PR. Feel free to merge without renaming if you want 🟢

Reply:
The condition is for if we should continue performing the watch loop. Right now we only end the watch loop for two reasons. If the bot is the only member or if the bot has left the chat for other reasons ( /leave command). We may want to add other conditions for ending the event loop. If anything the name should be tied to the continuation of the loop, not to a specific action that will cause the loop to end. If I didnt have to use setTimeout a while(continueLoop) would have made more sense... though with a while loop I would have gone with while(true) and used breaks to kill the loop.